src/cosmos/CosmosEngine.ts

```diff
@@ -520,38 +520,50 @@ export class CosmosEngine extends CurrencyEngine<
     ]
 
     for (const query of txQueryStrings) {
-      let startingPage = 1
-      let clients = this.getClients()
-
+      const clientsList: CosmosClients[] = []
       if (
-        this.networkInfo.archiveNode != null &&
+        this.networkInfo.archiveNodes != null &&
         Date.now() - TWO_WEEKS > this.otherData.archivedTxLastCheckTime
       ) {
-        startingPage = this.otherData[query].lastPage
-        clients = await createCosmosClients(
-          this.fetchCors,
-          rpcWithApiKey(this.networkInfo.archiveNode, this.tools.initOptions)
+        const sortedArchiveNodes = this.networkInfo.archiveNodes.sort(
+          (a, b) =>
+            a.blockTimeRangeSeconds.start - b.blockTimeRangeSeconds.start
         )
+        for (const node of sortedArchiveNodes) {
+          if (
+            node.blockTimeRangeSeconds.end == null ||
+            node.blockTimeRangeSeconds.end >
+              this.otherData.archivedTxLastCheckTime
+          ) {
+            const archiveClients = await createCosmosClients(
+              this.fetchCors,
+              rpcWithApiKey(node.endpoint, this.tools.initOptions)
+            )
+            clientsList.push(archiveClients)
+          }
+        }
       }
+      clientsList.push(this.getClients())
 
-      const { newestTxid, lastPage } = await this.queryTransactionsInner(
-        query,
-        clients,
-        startingPage
-      )
-      if (
-        (newestTxid != null &&
-          this.otherData[query]?.newestTxid !== newestTxid) ||
-        lastPage !== this.otherData[query]?.lastPage
-      ) {
-        this.otherData[query] = { newestTxid, lastPage }
-        this.walletLocalDataDirty = true
+      for (const clients of clientsList) {
+        const { newestTxid, lastTimestamp } = await this.queryTransactionsInner(
+          query,
+          clients
+        )
+        if (
+          newestTxid != null &&
+          this.otherData[query]?.newestTxid !== newestTxid
+        ) {
+          this.otherData[query] = { newestTxid }
+          this.otherData.archivedTxLastCheckTime = lastTimestamp
```

Comment:
Setting `archivedTxLastCheckTime` inside the for loop of txQueryStrings could cause the `receiver` query type to not execute since `archivedTxLastCheckTime` is global to both send and receive queries